feat/feature.go

```diff
@@ -67,6 +67,14 @@ type Range interface {
 	Len() int
 }
 
+// Feature is a Range whose coordinates are defined relative to a feature
+// location. Start and End return the coordinates of the feature relative to
+// its location which can be nil. In the latter case callers should make no
+// assumptions whether coordinates of such features are comparable.
+//
+// Name returns the name of the feature.
+// Description returns the description of the feature.
+// Location returns the reference feature where the feature is located.
 type Feature interface {
```

Comment:
Put these over the methods rather than in the top-level comment.

```
type Feature interface {
	Range

	// Name returns the name of the feature.
	Name() string

	// Description returns the description of the feature.
	Description() string

	// Location returns the reference feature on which
	// the feature is located.
	Location() Feature
}
```